packages/protocol/src/routers/ConnextRouter.sol

```diff
@@ -137,31 +148,25 @@ contract ConnextRouter is BaseRouter, IXReceiver {
     bytes memory callData
   )
     external
+    onlyXReceiveProxy
     returns (bytes memory)
   {
     (Action[] memory actions, bytes[] memory args) = abi.decode(callData, (Action[], bytes[]));
 
-    uint256 balance;
-    uint256 beforeSlipped;
-    if (amount > 0) {
-      // Ensure that at this entry point expected `asset` `amount` is received.
-      balance = IERC20(asset).balanceOf(address(this));
-      if (balance < amount) {
-        revert ConnextRouter__xReceive_notReceivedAssetBalance();
-      } else {
-        _tempTokenToCheck = Snapshot(asset, balance - amount);
-      }
+    IERC20 asset_ = IERC20(asset);
```

Comment:
maybe no need to declare an additional variable here, use directly `IERC20(asset)` in the next 2 lines